TrackerMC/inc/StrawDigiBundleCollection.hh

```diff
@@ -0,0 +1,139 @@
+// Containerize StrawDigiBundles, with a extra functionality for resolving
+// situations where multiple digis have overlapping digitization windows
+//
+// Ed Callaghan, 2023
+
+#ifndef TrackerMC_StrawDigiBundleCollection_hh
+#define TrackerMC_StrawDigiBundleCollection_hh
+
+// stl
+#include <map>
+#include <memory>
+#include <vector>
+
+// art
+#include <art/Framework/Principal/Handle.h>
+
+// cetlib_except
+#include <cetlib_except/exception.h>
+
+// mu2e
+#include <Offline/MCDataProducts/inc/StrawDigiMC.hh>
+#include <Offline/TrackerConditions/inc/StrawElectronics.hh>
+#include <Offline/TrackerMC/inc/StrawDigiBundle.hh>
+
+namespace mu2e{
+  using SDBC_iterator = std::vector<StrawDigiBundle>::iterator;
+  using SDBC_const_iterator = std::vector<StrawDigiBundle>::const_iterator;
+
+  class StrawDigiBundleCollection{
+    public:
+      // forwarded calls to underlying container
+      size_t size() const;
+      SDBC_iterator begin();
+      SDBC_const_iterator begin() const;
+      SDBC_iterator end();
+      SDBC_const_iterator end() const;
+      StrawDigiBundle& operator[](size_t i);
+      const StrawDigiBundle& operator[](size_t i) const;
+
+      // convenience methods for accepting new StrawDigiBundles
+      // from different source situations
+      template<template<typename> class Pointer>
+      void Append(const Pointer<StrawDigiCollection>& digis,
```

Comment:
I apologize if you described this in your talk and I missed it.  About the template parameter Pointer.  I think one possible value for that is vector - is that right?  What other values might it have?

Reply:
I did not address this in the slide-deck. `Pointer` here abstracts over different smart pointers. E.g. [preexisting digis](https://github.com/edcallaghan/mu2e-Offline/blob/straw_digi_mixing/TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc#L480) are accessed via an `art::Handle`, and newly created digis via [an stl class](https://github.com/edcallaghan/mu2e-Offline/blob/straw_digi_mixing/TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc#L553).

Reply:
When I look at the code, you are not using any of the features of the Handle/ValidHandle/etc inside this class.  In such a case our standard practice is to dereference handles when passing the payload to a function.   So the arguments become just StrawDigiCollection const&, etc.    The exception would be if you need to use features of the handles in the function; an example would be creating Ptrs to objects in the collections.

Reply:
Similarly we recommend that function arguments of geometry types be Tracker const& not GeomHandle<Tracker>.

Reply:
That practice makes sense to me. I've updated the signatures and calls to accept const refs with no templates.

Reply:
Thanks.